### PR TITLE
refactor(dfs): expose path, resolve error

### PR DESCRIPTION
**Pull request by koivunej**
Done in the past in closed #485 I re-did it because I didn't see I might need it in #521.

### CODE REVIEW

File: crates/pathfinder/src/state/merkle_tree.rs [hunk 1]
```diff
@@ -1443,11 +1441,11 @@ mod tests {
             uut.set(key.view_bits(), value).unwrap();
 
             let mut visited = vec![];
-            let mut visitor_fn = |node: &Node| {
+            let mut visitor_fn = |node: &Node, _: &_| {
                 visited.push(node.clone());
```
Comment (CHr15F0x) [inline crates/pathfinder/src/state/merkle_tree.rs:1445]: I'm wondering if it makes sense to assert for proper paths on the way too as in the original PR's tests.

**Review by CHr15F0x — APPROVED** [review]
LGTM

File: crates/pathfinder/src/state/merkle_tree.rs (hunk 1 above)
Comment (koivunej) [reply crates/pathfinder/src/state/merkle_tree.rs:1445]: Aah good point.

Comment (koivunej) [reply crates/pathfinder/src/state/merkle_tree.rs:1445]: I'll just cherry-pick your tests.

**Review by Mirko-von-Leipzig — COMMENTED** [review]
I don't actually know what this does.. :)

File: crates/pathfinder/src/state/merkle_tree.rs [hunk 2]
```diff
@@ -584,11 +584,13 @@ impl<T: NodeStorage> MerkleTree<T> {
     /// function can also return `ControlFlow::Break` to stop the visit with the given return
     /// value, which will be returned as `Some(value))` to the caller.
     ///
+    /// The bitslice represents the path to the node.
+    ///
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/state/merkle_tree.rs:588]:
```suggestion
    /// The visitor function receives the node being visited, as well as the full path to that node.
    ///
```

**Pull request merged**